src/routers/resource_routers/organisation_router.py

```diff
@@ -18,3 +38,149 @@ def resource_name_plural(self) -> str:
     @property
     def resource_class(self) -> type[Organisation]:
         return Organisation
+
+    def add_custom_routes(self, router: APIRouter, path: str):
+        @router.post(path, tags=["organisations"])
+        async def organisation_image(
+            identifier: str,
+            file: UploadFile = File(...),
+            name: str = Query(..., description="Uploaded image filename", example="logo"),
+            session=Depends(get_session),
+            user: KeycloakUser | None = Depends(get_user_or_raise),
+        ):
+            validate_image_type(file)
+
+            org = session.exec(
+                select(Organisation).where(Organisation.identifier == identifier)
+            ).one_or_none()
+
+            if not org:
+                raise HTTPException(
+                    status_code=HTTPStatus.NOT_FOUND,
+                    detail=f"Organisation {identifier} not found in the database.",
+                )
+            # Donot allow image upload with same name.
+            # We do not check for identical image content (only name).
+            # Consider adding a limit on the number of uploaded images in the future.
+
+            existing_media = next((m for m in org.media if m.name == name), None)
+            if existing_media:
```

Comment:
```suggestion
            if any(media.name == name for media in org.media):
```
Previous code is fine, but just wanted to give an alternative for consideration that I feel better signals the intent. It reads much closer to the comment you wrote above (which may then be considered superfluous?).

Reply:
Yeah, good suggestion.